modules/gateway/controllers/routeset/populate/virtualservice.go

```diff
@@ -66,16 +68,14 @@ func virtualServiceFromRoutesets(systemNamespace string, clusterDomain *projectv
 				port = int(*dest.Port)
 			}
 
-			if i == len(routeSpec.To)-1 && constants.ServiceMeshMode == constants.ServiceMeshModeLinkerd {
-				host := dest.Service
-				if dest.Revision != "" {
-					host = host + "-" + dest.Revision
+			if constants.ServiceMeshMode == constants.ServiceMeshModeLinkerd {
+				httpRoute.Headers.Request.Set["l5d-dst-override"] = fmt.Sprintf("%s-%v.%s.svc.cluster.local:%v", routeSet.Name, index, routeSet.Namespace, port)
```

Comment:
I feel like we need to mention something about https://github.com/linkerd/linkerd2-proxy/pull/173
Or document what `l5d-dst-override` does. I'm even fine with a link the linkerd2 PR. I feel like this is hard to understand without some more context with a comment